DependencyInjection/DoctrineMongoDBExtension.php

```diff
@@ -77,6 +79,13 @@ public function load(array $configs, ContainerBuilder $container)
 
         // BC Aliases for Document Manager
         $container->setAlias('doctrine.odm.mongodb.document_manager', new Alias('doctrine_mongodb.odm.document_manager'));
+
+
+        $container->registerForAutoconfiguration(ServiceDocumentRepositoryInterface::class)
+            ->addTag(ServiceRepositoryCompilerPass::REPOSITORY_SERVICE_TAG);
+
+        $container->getDefinition('doctrine_mongodb.odm.container_repository_factory')
```

Comment:
Ooof. I can't remember.

Here is my guess. This feature only works with 3.3 and higher... because you need the `ServiceLocatorTagPass`. I *think* (not sure) that I was trying to avoid a nasty error in 3.2. Specifically, if you set the `ServiceLocator` into your XML file, that would fail on 3.2 - as that class doesn't exist. However, I'm not sure why I didn't just leave that argument blank. 

Look at the original commit : https://github.com/doctrine/DoctrineBundle/commit/4efcb135344e55403a0fd193c23a8c7249cfd7d0#diff-8f3578ddca2c59937b37c3331b2ab336R388

I set the first argument to null in XML (https://github.com/doctrine/DoctrineBundle/commit/4efcb135344e55403a0fd193c23a8c7249cfd7d0#diff-8f3578ddca2c59937b37c3331b2ab336R388) to avoid errors on 3.2 I believe. But... I'm not sure why I set this to an empty ServiceLocator in the extension class. I'm *guessing* that there was something that was *forcing* me to have this properly set. Perhaps a type-check on my `null` versus the type-hint on that argument. That would make sense :). 

If you're not supporting Symfony 3.2... I'm not sure any of this is necessary. Here's an example from core - it looks super simple: https://github.com/symfony/symfony/blob/f834c9262b411aa5793fcea23694e3ad3b5acbb4/src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/RuntimeLoaderPass.php#L37 and the definition: https://github.com/symfony/symfony/blob/f834c9262b411aa5793fcea23694e3ad3b5acbb4/src/Symfony/Bundle/TwigBundle/Resources/config/twig.xml#L159

Try that simpler route - see if there are any issues.

Reply:
I’ll give that a shot, thank you for your help!

Reply:
Dropping `null` as well as the `replaceArgument` call in the extension works. Not sure about Symfony < 3.4, but we don't use those here.